exercises/simple-cipher/canonical-data.json

```diff
@@ -0,0 +1,145 @@
+{
+  "exercise": "simple-cipher",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "Random key cipher",
+      "cases": [
+        {
+          "description": "Key is made only of lowercase letters",
+          "property": "key",
+          "input": {},
+          "expected": {
+            "match": "^[a-z]+$"
+          }
+        },
+        {
+          "description": "It can encode",
+          "property": "encode",
+          "input": {
+            "plaintext": "aaaaaaaaaa"
+          },
+          "expected": "cipher.key"
+        },
+        {
+          "description": "It can decode",
+          "property": "decode",
+          "input": {
+            "ciphertext": "cipher.key"
+          },
+          "expected": "aaaaaaaaaa"
+        },
+        {
+          "description": "It is reversible. I.e., if you apply decode in a encoded result, you must see the same plaintext encode parameter as a result of the decode method",
+          "property": "decode",
+          "input": {
+            "plaintext": "abcdefghij",
+            "ciphertext": "cipher.encode"
+          },
+          "expected": "abcdefghij"
+        }
+      ]
+    },
+    {
+      "description": "Incorrect key cipher",
+      "cases": [
+        {
+          "description": "It throws an error with an all uppercase key",
+          "property": "new",
```

Comment:
This is interesting since `new` might imply too much an object-oriented interface, which would raise questions as to how this file is to be understood by languages less amenable to object-oriented interfaces.

I don't have a concrete suggestion yet. Let's see if there are any.

Reply:
Good point, this is a tricky one.

I think the problem is that we're not testing the constructor method itself, but verifying that assigning a invalid key to `key` must cause a error, what is done in the constructor here. 

One option for me would be set the property to `key` or maybe something like `key=` will be also valid for me. I don't have any better/concrete suggestion too.

Reply:
We could also use the `encode` property and name the case something like `Encode using uppercase key`.